httpcore5/src/test/java/org/apache/hc/core5/http/io/entity/TestPathEntity.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.core5.http.io.entity;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+
+import org.apache.hc.core5.http.ContentType;
+import org.junit.Assert;
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link PathEntity}.
+ */
+public class TestPathEntity {
+
+    @Test
+    public void testBasics() throws Exception {
+        final Path tmpPath = Files.createTempFile("testfile", ".txt");
+        tmpPath.toFile().deleteOnExit();
+        try (final PathEntity httpentity = new PathEntity(tmpPath, ContentType.TEXT_PLAIN)) {
+            Assert.assertEquals(Files.size(tmpPath), httpentity.getContentLength());
```

Comment:
How does this test make sense if the file is empty?

Reply:
Added comment.